Integration of GitHub Actions

alonweissfeld opened the pull request:
Implemented GitHub Actions to our project:

As configured in `.github/workflows/python-package.yml`, using ubuntu machine, we're:
- Setting up python (version `3.8.x`)
- Installing necessary dependencies (`flake8`, `pipenv`)
- Running flake8 lint on all python files.
- Running the Django tests on the virtual environment (after setting up the relevant packages with pipenv and migrating the models).

In addition, there was a bug in the migration files - I've removed the old ones and created a new file.
Finally, this required to manually lint the failed `flake8` files.

Thanks

Commit:
Added a 'python-package' to intergrate GitHub Actions.
The yml file determines the following actions to preform:
- Setting up the python version and relevnat dependencies (flake8, pipenv)
- Running flake8 to lint all python files
- Migrating the db and running the django tests inside the virtual environmnet.

Also, I've replaced the migration files to a new fresh one - there was a bug with the multiple migrations files (some were dependencies and were deleted).

Commit:
Applied manual flake8 linting on all necessary files.

Commit:
Add the workflow activation on pull requests.

11galpid reviewed (approved):
Looks good!
General comments:
1. Maybe you should separate the PR into three-PR's, one for fixing the migration files, second for 'linting' the files, and third for adding the CI workflow.
2. The third action 'Run django tests' should be separated further, to (1) running pipenv sync, (2) running migrations (3) running the test for the main_app. This way if one fails, its easier to map the problem.

To be frank, I'm fine with leaving this as one PR, it's just by the way we worked so far, I feel that's the approach the Red Hatters would like us to go..
In any case great job, thumbs up

11galpid commented:
After going over more PR's I see the 'pylint' issue repeats - we need to address it globally in a different PR.
I created a separate issue for that, but as the issue blocks this PR from pushing the CI in, maybe we should include the required parts of it in this PR for faster inclusion of the CI's.

Commit:
Map the tests action to separate related tasks.

alonweissfeld commented:
@11galpid I've split this PR to three ones (#56, #57, #58). I believe it's the right way to go - we can merge the linting and migration PR's before to avoid them as blockers. Take a look please on the Actions PR - I've addressed your comment on the `Run django tests` action :)
